internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -116,6 +117,13 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 	}
 
+	if reconcileManagedLabels(machine, node) {
+		options := []client.PatchOption{
+			client.FieldOwner("MachineController"),
+		}
+		return ctrl.Result{}, remoteClient.Patch(ctx, node, client.Apply, options...)
```

Comment:
Please note that in the Node object there should be only the fields that we care about with SSA. With the current approach we take ownership of all previously existing fields.

Essentially we have to build the Node object from scratch and only add the fields we care about + metadata.{name,namespace,uid} (we also set apiVersion + kind in the topology controller, but I think that is only required with unstructured)

Name is needed to identify the object. uid is a safety that we just update an existing node (SSA would automatically create a new node if uid is empty and no Node with the right name exists)

Reply:
@sbueringer we need to make sure our handcrafted Node has no additional empty values i.e. .spec, for that we need to handcraft a Node type as in k8s.io/client-go/applyconfigurations/ and if we want to reuse controller runtime client we need to make sure it implements client.Object{}, correct?

Should we just use unstructured instead?

Reply:
Yes, let's use unstructured

Reply:
updated.